### PR TITLE
DOV-6867: Adapt Settings Framework

### PR DESCRIPTION
**Pull request by fleischie**
Necessary changes to the core project  to support the config-rewrite changes.

### CODE REVIEW

**Comment by sirainen** [comment]
@slusarz Are you reviewing this?.. I have at least no idea.

**Comment by slusarz** [comment]
> @slusarz Are you reviewing this?.. I have at least no idea.

@sirainen : I can.  But kind of waiting for @fleischie to inform it's ready to be looked at :)  Since it seems like commits are still trickling in.

File: components/SettingsComponent.vue [hunk 1]
```diff
@@ -51,6 +51,10 @@ const d = Object.fromEntries(Object.entries(data).filter(([k, v]) =>
   color: var(--vp-custom-block-danger-text);
   background-color: var(--vp-custom-block-danger-bg);
 }
+
+th {
```
Comment (cmouse) [inline components/SettingsComponent.vue:55]: can this be
```
.dovecotSettings th
```
please?

**Review by cmouse — APPROVED** [review]

**Review by cmouse — CHANGES_REQUESTED** [review]
one nit

File: components/SettingsComponent.vue (hunk 1 above)
Comment (fleischie) [reply components/SettingsComponent.vue:55]: It can, yes.
But the style is scoped, so it wouldn't leak to other components anyway.

**Comment by fleischie** [comment]
> one nit

Timo reprimanded and I suggested the "Default" row in settings tables. Which means I will maybe try to find a better solution for this. Also maybe add another setting `IN_PORT` to specify unsigned integer values that are max. `65535`.

File: util/generate_man.js [hunk 2]
```diff
@@ -122,7 +122,7 @@ const processDovecotMd = () => {
 const main = async (component, outPath) => {
 	/* Create output directory, if it doesn't exist. */
 	if (!fs.existsSync(outPath)) {
-		await fs.promises.mkdir(outPath)
+		await fs.promises.mkdir(outPath, { recursive: true })
```
[the remark this answers is not in the document]
Comment (fleischie) [reply util/generate_man.js:125]: I don't see the benefit, tbh. As per the documentation using `{ recursive: true }` will not yield an error if the directory already exists.

File: lib/data/events.data.js [hunk 3]
```diff
@@ -62,6 +62,22 @@ function normalizeEventsFields(fields) {
 	return f
 }
 
+/* Return list of added/changed/deprecated/removed items, this can optionally
```
[the remark this answers is not in the document]
Comment (fleischie) [reply lib/data/events.data.js:65]: I forgot that the functional changes were added in another, previous PR. I rewrote the title.

**Comment by fleischie** [comment]
@cmouse nit addressed, follow-up comment also resolved
@slusarz maybe you want to chime in as well? you may have additional opinions on the system changes.

File: components/SettingsComponent.vue [hunk 4]
```diff
@@ -85,6 +85,14 @@ const d = Object.fromEntries(Object.entries(data).filter(([k, v]) =>
       <span class="comma" v-for="v in v.values_enum.values()"><code>{{ v }}</code></span>
      </td>
     </tr>
+    <tr v-if="v.dependencies.length">
+     <th style="text-align:right;">Dependencies</th>
```
Comment (cmouse) [inline components/SettingsComponent.vue:89]: you forgot to remove the text-align here.

**Comment by cmouse** [comment]
LGTM, one nit and then we can merge this.

**Comment by fleischie** [comment]
> LGTM, one nit and then we can merge this.

Nit resolved. The extra commits were because I forgot that I needed to rebase unto `feature/config-rewrite`, not `main`.

**Pull request merged**